NoteFind/Views/NoteCell.m

```diff
@@ -24,9 +35,33 @@ - (void)setNote:(Note *)note {
     _note = note;
     _postTitle.text = _note.title;
     _postCaption.text = _note.noteDescription;
-    self.postUsername.text = _note.author.username;
+    _postUsername.text = _note.author.username;
     _postNote.file = _note.note;
     [_postNote loadInBackground];
+    
+    // TODO: add addCount
+}
+- (IBAction)didTapDownload:(id)sender {
+    NSManagedObject *newOfflineNote = [NSEntityDescription insertNewObjectForEntityForName:@"OfflineNote" inManagedObjectContext:context];
+    NSData *fileData = [_postNote.file getData];
+    if (fileData == nil) {
+        [self.delegate showErrorMessage];
+    } else {
+        [newOfflineNote setValue:_postUsername.text forKey:@"username"];
+        [newOfflineNote setValue:_postTitle.text forKey:@"title"];
+        [newOfflineNote setValue:_postCaption.text forKey:@"caption"];
+        [newOfflineNote setValue:fileData forKey:@"noteFileData"];
+        [newOfflineNote setValue:_postNote.file.name forKey:@"noteFileName"];
+        [appDelegate saveContext];
+        
+    }
+    // TODO: add addCount
+    
+    // TODO: fetch data
+    
+    NSFetchRequest *request = [NSFetchRequest fetchRequestWithEntityName:@"OfflineNote"];
+    NSArray *results = [context executeFetchRequest:request error:nil];
+    NSLog(@"results: %@", results);
```

Comment:
What are these past few lines used for? Seems like debugging - if so lets remove it from your commiit